src/remote.c

```diff
@@ -334,5 +379,11 @@ int dcc_compile_remote(char **argv,
         dcc_collect_child("ssh", ssh_pid, &ssh_status, timeout_null_fd); /* ignore failure */
     }
 
+    if (ret == 0 && *status != 0) {
+        if ((ret = dcc_check_unsupported_directives(cpp_fname, input_fname))) {
```

Comment:
It looks like check happens after attempting to compile it remotely fails. From the PR description I would have thought you could detect this first, and save a bit of time?

Reply:
> It looks like check happens after attempting to compile it remotely fails.

Correct.

> From the PR description I would have thought you could detect this first, and save a bit of time?

That's what I did in the first version. Reportedly it slows down the build sometimes (see the comment by @wtarreau).

Files with asm `.incbin` directives are very rare, like 1 in 10000 (in Linux kernel there might be a handful of them, depending on config).  On the other hand preprocessed C sources are rather long (several MB). Scanning ~ 10000 such files might be slower than running an extra remote compilation.

Reply:
Ah, OK. Could you add that explanation in a comment?

Reply:
Updated the commit message and description of the pull request.